CONTRIBUTING.md

```diff
@@ -383,6 +383,37 @@ tested it becomes an officially supported Akka feature.
 
 [List of Akka features marked as may change](http://doc.akka.io/docs/akka/current/common/may-change.html)
 
+## Java APIs
+
+Each feature in Akka must have an API for convenient usage from both Scala and Java, as we develop the Java APIs in
+a subset of Scala that is easy to use from Java there are ... 
+
+When new APIs are added or existing ones are significantly changed such changes should have test coverage written in 
+actual Java code.
+
+New APIs are mostly separated into a `scaladsl` and a `javadsl` subtree of packages to discern what APIs are for which 
+language. In a few cases common types are shared between the two languages, but the general rule should be the separate
+packages style. Some of the existing APIs do not follow this pattern for historical reasons and it would be too big a 
+change (especially given our binary compatibility guarantees) to introduce it (the untyped `akka.actor.ActorSystem` for 
+example).
```

Comment:
Perhaps mention 'internal' as a package name for API-agnostic code, possibly implementing or being called from the javadsl/scaladsl packages?